### ISSUE TITLE
ForbiddenClassName should allow exact or regex matching

### ISSUE THREAD

**Issue by ajyoon** [detekt/detekt#8406]
## Expected Behavior

I would like to forbid certain class names by _exact_ match. For example, in my codebase I just tried to create this rule to prevent folks from shadowing common library classes:

```yaml
naming:
  ForbiddenClassName: # Forbid shadowing common type names
    active: true
    forbiddenName:
      - 'LocalDate'
      - 'YearMonth'
      - 'Money'
```

Alternatively, I would expect regexp matchers indicating full string matches should do this:

```yaml
naming:
  ForbiddenClassName: # Forbid shadowing common type names
    active: true
    forbiddenName:
      - '^LocalDate$'
      - '^YearMonth$'
      - '^Money$'
```

## Current Behavior

To my surprise, the first rule matches any class name with these names as _substrings_, and the second rule treats the `^$` symbols as literals.

## Context

This limits the usefulness of the `ForbiddenClassName` feature for me, since I want to allow names which _include_ these strings, but not names which are _exactly_ them. In general, I would expect that name matching rules are either exact string matches, or support basic globbing, or are full-blown regexps. The behavior where the given names are unconditionally treated as substrings is surprising to me.

In any case, I think it would be helpful to improve the documentation around this lint, since currently the docs don't actually say how the class names are specified.

https://detekt.dev/docs/rules/naming/#forbiddenclassname

> ForbiddenClassName
>
>Reports class names which are forbidden per configuration. By default, this rule does not report any classes. Examples for forbidden names might be too generic class names like ...Manager.
>
> Active by default: No
>
>Debt: 5min
> Configuration options:
>
>    forbiddenName (default: [])
>
>    forbidden class names

**Comment by marschwar** [comment]
I totally agree. This behavior is rather surprising. In many other places we support basic wildcards using `*` and `?`. I think this would be totally sufficient here. 

But this would change the current behavior and therefor be a breaking change. I would vote to change the rule in 2.x.

Any other opinions?

**Comment by BraisGabin** [comment]
+1. This is a great moment to make this change. 2.0 allow us to fix these kind of issues.

@ajyoon do you want to give it a try?

**Comment by ajyoon** [comment]
I just opened a PR :)
